src/System.Private.CoreLib/shared/System/NotFiniteNumberException.cs

```diff
@@ -55,13 +55,13 @@ public NotFiniteNumberException(string message, double offendingNumber, Exceptio
 
         protected NotFiniteNumberException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
-            _offendingNumber = info.GetInt32("OffendingNumber");
+            _offendingNumber = info.GetDouble("OffendingNumber"); // Do not rename (binary serialization)
```

Comment:
Is there any way this could cause a problem?  Maybe with some custom IFormatterConverter that didn't allow for an Int32 => Double conversion?  It's probably fine, just want to double-check.

Reply:
Yes, I thought about that too. If somebody is using a custom FormatterConverter and has a throwing implementation for https://source.dot.net/#System.Private.CoreLib/shared/System/Runtime/Serialization/IFormatterConverter.cs,23 than this will cause errors. That said, I think this is a safe change.

Reply:
This will have an observable impact on behavior, but arguably it's a bug fix, e.g. on netfx, this:
```C#
using System;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

class Program
{
    static void Main()
    {
        var bf = new BinaryFormatter();
        var s = new MemoryStream();

        var e = new NotFiniteNumberException(12.34);
        Console.WriteLine(e.OffendingNumber);

        bf.Serialize(s, e);
        s.Position = 0;

        e = (NotFiniteNumberException)bf.Deserialize(s);
        Console.WriteLine(e.OffendingNumber);
    }
}
```
outputs:
```
12.34
12
```
whereas after this fix it'll output:
```
12.34
12.34
```
as you'd expect.

Reply:
Thanks for investigating further! I'm a little bit surprised that this was never reported as it's the same in netfx. Probably a not very often serialized exception.

The tests didn't cover this because we only added equality extension tests for Exception and AggregatedException: https://source.dot.net/#System.Runtime.Serialization.Formatters.Tests/EqualityExtensions.cs,1178

Adding equality tests for all other exception types wasn't feasible cost-wise when we backported the exception serialization feature. Right now all exception types are equality checked but only for the base properties: https://source.dot.net/#System.Runtime.Serialization.Formatters.Tests/EqualityExtensions.cs,1107

We should at some point add the missing equality checks. cc @danmosemsft